lib/main.dart

```diff
@@ -128,125 +112,50 @@ class _MyAppState extends State<MyApp> {
         body: Center(
           child: isBusy
               ? const CircularProgressIndicator()
-              : isLoggedIn
-                  ? Profile(logoutAction, name, picture)
+              : _credentials != null
+                  ? Profile(logoutAction, _credentials?.user)
                   : Login(loginAction, errorMessage),
         ),
       ),
     );
   }
 
-  Map<String, Object> parseIdToken(String idToken) {
-    final List<String> parts = idToken.split('.');
-    assert(parts.length == 3);
-
-    return jsonDecode(
-        utf8.decode(base64Url.decode(base64Url.normalize(parts[1]))));
-  }
-
-  Future<Map<String, Object>> getUserDetails(String accessToken) async {
-    const String url = 'https://$AUTH0_DOMAIN/userinfo';
-    final http.Response response = await http.get(
-      url,
-      headers: <String, String>{'Authorization': 'Bearer $accessToken'},
-    );
-
-    if (response.statusCode == 200) {
-      return jsonDecode(response.body);
-    } else {
-      throw Exception('Failed to get user details');
-    }
-  }
-
   Future<void> loginAction() async {
     setState(() {
       isBusy = true;
       errorMessage = '';
     });
 
     try {
-      final AuthorizationTokenResponse result =
-          await appAuth.authorizeAndExchangeCode(
-        AuthorizationTokenRequest(
-          AUTH0_CLIENT_ID,
-          AUTH0_REDIRECT_URI,
-          issuer: 'https://$AUTH0_DOMAIN',
-          scopes: <String>['openid', 'profile', 'offline_access'],
-          // promptValues: ['login']
-        ),
-      );
-
-      final Map<String, Object> idToken = parseIdToken(result.idToken);
-      final Map<String, Object> profile =
-          await getUserDetails(result.accessToken);
-
-      await secureStorage.write(
-          key: 'refresh_token', value: result.refreshToken);
+      final credentials = await auth0.webAuthentication(scheme: 'flutterdemo').login();
 
       setState(() {
         isBusy = false;
-        isLoggedIn = true;
-        name = idToken['name'];
-        picture = profile['picture'];
+        _credentials = credentials;
       });
     } on Exception catch (e, s) {
       debugPrint('login error: $e - stack: $s');
 
       setState(() {
         isBusy = false;
-        isLoggedIn = false;
         errorMessage = e.toString();
       });
     }
   }
 
   Future<void> logoutAction() async {
-    await secureStorage.delete(key: 'refresh_token');
+    await auth0.webAuthentication(scheme: 'flutterdemo').logout();
```

Comment:
Maybe we can extract `flutterdemo` to a constant?

Reply:
Sure thing!